gradle-enterprise-maven-extension-quarkus/src/main/java/com/gradle/QuarkusBuildCache.java

```diff
@@ -0,0 +1,327 @@
+            inputs.fileSet("quarkusCompileClasspath", compileClasspathElements, fileSet -> fileSet.normalizationStrategy(MojoMetadataProvider.Context.FileSet.NormalizationStrategy.CLASSPATH));
+        } catch (DependencyResolutionRequiredException e) {
+            throw new IllegalStateException("Classpath can't be resolved");
+        }
+    }
+
+    private void addMojoInputs(MojoMetadataProvider.Context.Inputs inputs) {
+        inputs
+            .fileSet("generatedSourcesDirectory", fileSet -> {
+            })
+            .properties("appArtifact", "closeBootstrappedApp", "finalName", "ignoredEntries", "manifestEntries", "manifestSections", "skip", "skipOriginalJarRename", "systemProperties", "properties")
+            .ignore("project", "buildDir", "mojoExecution", "session", "repoSession", "repos", "pluginRepos", "attachRunnerAsMainArtifact", "bootstrapId", "buildDirectory");
+    }
+
+    private void addQuarkusPropertiesInput(MojoMetadataProvider.Context.Inputs inputs, QuarkusExtensionConfiguration extensionConfiguration) {
+        inputs.fileSet("quarkusConfigCheck", new File(extensionConfiguration.getCurrentConfigFileName()), fileSet -> fileSet.normalizationStrategy(MojoMetadataProvider.Context.FileSet.NormalizationStrategy.RELATIVE_PATH));
+        inputs.fileSet("generatedSourcesDirectory", fileSet -> {
+        });
+    }
+
+    private void addQuarkusFilesInputs(MojoMetadataProvider.Context.Inputs inputs, Properties quarkusCurrentProperties) {
+        for (String quarkusFilePropertyKey : QUARKUS_KEYS_AS_FILE_INPUTS) {
+            String quarkusFilePropertyValue = quarkusCurrentProperties.getProperty(quarkusFilePropertyKey);
+            if (isNotEmpty(quarkusFilePropertyValue)) {
+                inputs.fileSet(quarkusFilePropertyKey, new File(quarkusFilePropertyValue), fileSet -> fileSet.normalizationStrategy(MojoMetadataProvider.Context.FileSet.NormalizationStrategy.RELATIVE_PATH));
+            }
+        }
+    }
+
+    private boolean isNotEmpty(String value) {
+        return value != null && !value.isEmpty();
+    }
+
+    private void configureOutputs(MojoMetadataProvider.Context context) {
+        context.outputs(outputs -> {
+            String quarkusExeFileName = "target/" + context.getProject().getBuild().getFinalName() + "-runner";
+            String quarkusJarFileName = "target/" + context.getProject().getBuild().getFinalName() + ".jar";
+            String quarkusUberJarFileName = "target/" + context.getProject().getBuild().getFinalName() + "-runner.jar";
+
+            outputs.cacheable("this plugin has CPU-bound goals with well-defined inputs and outputs");
+            outputs.file("quarkusExe", quarkusExeFileName);
+            outputs.file("quarkusJar", quarkusJarFileName);
+            outputs.file("quarkusUberJar", quarkusUberJarFileName);
+
+            // Do not declare dump config as goal output to avoid
```

Comment:
should we remove this? or is there a case where it is needed?

Reply:
It was kept here for documentation purpose but we can indeed remove it to clarify 👍